RFC/src/RFC-0201_TariScript.md

```diff
@@ -524,6 +524,10 @@ $$
 For Bob's part, when he discovers one-sided payments to himself, he should spend them to new outputs using a traditional
 transaction to thwart any potential horizon attacks in the future.
 
+One other problem with one sided communication is the communication of the value of the transaction being sent, and the wallet recovery of one sided payments. We can solve both of these by using (\\( k_{Ob} \\).
+Because this is a secret key known by Alice and Bob and no one else. We hash this key as a new key, (\\( k_{rewind} =  Hash(k_{Ob}) \\). We know have a rewinding secret key for use in bulletproof rewinding. And now
+we can encrypt the value (\\( v \\) inside of the bulletproof rewinding, allowing Bob to unlock his payment without revealing any information publicly.
+
```

Comment:
I would add something like this:

-----------

In summary, receiving party scanning the blockchain for one-sided payments therefore involves:
- Scan all UTXOs trying to unlock the payment script hashes; if the script resolves when inserting the receiver public key, it is for the receiver.
- With those identified UTXOs, do Bulletproof rewinding with the range of registered sender public rewind keys ((\( k_{rewind}G \) as per above). If the proof message matches, the rewound value is more than likely the correct value of the commitment.
- Test opening of the commitment using the value obtained from rewinding and the spending key from the Diffie-Hellman exchange; if the commitment can be opened, it is intended for the receiver.

Reply:
Don't agree with point 2. Point 2 should be rewound using the hash of the blinding factor

Reply:
We equate the private rewind key to the hash of the blinding factor. Bulletproof rewinding is done with the public rewind key, not the private rewind key.